gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -3359,14 +3390,31 @@ MacroExpander::expand_crate ()
   for (auto it = items.begin (); it != items.end ();)
     {
       auto &item = *it;
+      rust_debug ("expanding item %ld", it - items.begin ());
 
       // mark for stripping if required
       item->accept_vis (attr_visitor);
 
-      if (item->is_marked_for_strip ())
+      auto fragment = take_expanded_fragment ();
```

Comment:
Is this where we expand the invocations SingleAstNodes?

It seems like a good idea actually, though I am having trouble following the logic here. I would have assumed we could do something like:

```
for (auto& item : crate.items) {
  if (item.is_macro_invcoation()) {
    <take ownership of all fragments>
    <append-fragments to crate>
    <clear-fragments from macro-invocation>
  }
}
```